apps/sms/services/js/moz_mobile_message/moz_mobile_message_shim.js

```diff
@@ -0,0 +1,210 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+
+/* global bridge, Utils */
+
+/* exported MozMobileMessageShim */
+
+'use strict';
+(function(exports) {
+
+/**
+ * Name of the service for mozMobileMessage API shim
+ * @type {string}
+ */
+
+const SERVICE_NAME = 'mozMobileMessageShim';
+
+/**
+ * Array of the event names that is corresponding to mozMobileMessage events.
+ * @type {Array.<string>}
+ */
+
+const EVENTS = ['received', 'sending', 'sent', 'failed', 'deleted',
+                'readsuccess', 'deliverysuccess'];
+
+/**
+ * Array of method names that need to be exposed for API shim.
+ * @type {Array.<string>}
+ */
+
+const METHODS = ['getMessage', 'retrieveMMS', 'send', 'sendMMS',
+                 'delete', 'markMessageRead', 'getSegmentInfoForText'];
+
+/**
+ * Array of stream names that need to return data (messages/threads) in chunk.
+ * @type {Array.<string>}
+ */
+
+const STREAMS = ['getThreads', 'getMessages'];
+
+const mozMobileMessage = navigator.mozMobileMessage;
+
+var service = null;
+
+var MozMobileMessageShim = {
+  init: function ms_init() {
```

Comment:
nit: I like that @julienw started to use new ```init() {}``` notation instead, you can use it here as well if you like it :) But up to you since it's shim that should eventually go away.

Also just theoretical question (not for this PR): should we pass ```navigator.mozMobileMessage or other navigator.mozApi``` as init parameter for such shims?

This likely will make unit tests easier, I mean we can get rid of these 
```
realMozMobileMessage = navigator.mozMobileMessage;
navigator.mozMobileMessage = MockNavigatormozMobileMessage;
```
and just use
```
setup(function() {
  shim.init(MockNavigatormozMobileMessage)
});
```